### PR TITLE
[actpool] validation for blobtx

### PR DESCRIPTION
**Pull request by envestcc**
# Description

There are many differences between Blob-tx and regular tx, requiring different validation rules to ensure the correctness of transactions while maintaining program performance.

Based on #4366 

## Type of change
Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by CoderZhi — APPROVED** [review]

File: actpool/actpool.go [hunk 1]
```diff
@@ -162,12 +162,26 @@ func (ap *actPool) Start(ctx context.Context) error {
 	if ap.store == nil {
 		return nil
 	}
+	// open action store and load all actions
+	blobs := make(SortedActions, 0)
 	err := ap.store.Open(func(selp *action.SealedEnvelope) error {
+		isBlobTx := len(selp.BlobHashes()) > 0
+		if isBlobTx {
```
Comment (dustinxie) [inline actpool/actpool.go:169]: `if len(selp.BlobHashes()) > 0`

File: actpool/actqueue.go [hunk 2]
```diff
@@ -106,6 +109,30 @@ func (q *actQueue) Put(act *action.SealedEnvelope) error {
...
+				return errors.Wrap(action.ErrReplaceUnderpriced, "blob tx can only replace blob tx")
+			}
+			var (
+				prevBlobGasFeeCap = actInPool.BlobGasFeeCap()
+				blobGasFeeCap     = act.BlobGasFeeCap()
+
+				priceBump        = big.NewInt(2)
+				minGasFeeCap     = new(big.Int).Mul(actInPool.GasFeeCap(), priceBump)
+				minGasTipCap     = new(big.Int).Mul(actInPool.GasTipCap(), priceBump)
+				minBlobGasFeeCap = new(big.Int).Mul(prevBlobGasFeeCap, priceBump)
```
Comment (dustinxie) [inline actpool/actqueue.go:125]: no need to define `prevBlobGasFeeCap` at L119, directly use `actInPool.BlobGasFeeCap()` here?

File: actpool/actqueue.go [hunk 3]
```diff
@@ -106,6 +109,30 @@ func (q *actQueue) Put(act *action.SealedEnvelope) error {
...
+			)
+			switch {
+			case act.GasFeeCap().Cmp(minGasFeeCap) < 0:
+				return errors.Wrapf(action.ErrReplaceUnderpriced, "gas fee cap %s < %s", act.GasFeeCap(), minGasFeeCap)
+			case act.GasTipCap().Cmp(minGasTipCap) < 0:
+				return errors.Wrapf(action.ErrReplaceUnderpriced, "gas tip cap %s < %s", act.GasTipCap(), minGasTipCap)
+			case blobGasFeeCap.Cmp(minBlobGasFeeCap) < 0:
```
Comment (dustinxie) [inline actpool/actqueue.go:132]: no need to define `blobGasFeeCap` at L120, directly use `act.BlobGasFeeCap()` here?

File: actpool/actqueue.go [hunk 4]
```diff
@@ -184,10 +219,14 @@ func (q *actQueue) cleanTimeout() []*action.SealedEnvelope {
 		nonce := q.ascQueue[i].nonce
 		if timeNow.After(q.ascQueue[i].deadline) && nonce > q.pendingNonce {
 			removedFromQueue = append(removedFromQueue, q.items[nonce])
+			isBlobTx := len(q.items[nonce].BlobHashes()) > 0
 			delete(q.items, nonce)
 			delete(q.pendingBalance, nonce)
 			q.ascQueue[i] = q.ascQueue[size-1]
 			size--
+			if isBlobTx {
```
Comment (dustinxie) [inline actpool/actqueue.go:227]: `if len(q.items[nonce].BlobHashes()) > 0`

File: actpool/actqueue.go [hunk 5]
```diff
@@ -220,7 +259,11 @@ func (q *actQueue) UpdateAccountState(nonce uint64, balance *big.Int) []*action.
 		heap.Remove(&q.descQueue, nttl.descIdx)
 		nonce := nttl.nonce
 		removed = append(removed, q.items[nonce])
+		isBlobTx := len(q.items[nonce].BlobHashes()) > 0
 		delete(q.items, nonce)
+		if isBlobTx {
```
Comment (dustinxie) [inline actpool/actqueue.go:264]: same as above

File: actpool/actqueue.go [hunk 6]
```diff
@@ -337,6 +381,9 @@ func (q *actQueue) PopActionWithLargestNonce() *action.SealedEnvelope {
 	item := q.items[itemMeta.nonce]
 	delete(q.items, itemMeta.nonce)
 	q.updateFromNonce(itemMeta.nonce)
-
+	isBlobTx := len(item.BlobHashes()) > 0
+	if isBlobTx {
```
Comment (dustinxie) [inline actpool/actqueue.go:385]: same here

File: actpool/queueworker.go [hunk 7]
```diff
@@ -180,6 +180,19 @@ func (worker *queueWorker) checkSelpWithState(act *action.SealedEnvelope, pendin
 		return action.ErrNonceTooHigh
 	}
 
+	// Nonce must be continuous for blob tx
+	isBlobTx := len(act.BlobHashes()) > 0
+	if isBlobTx {
```
Comment (dustinxie) [inline actpool/queueworker.go:185]: same here

**Review by dustinxie — APPROVED** [review]

**Pull request merged**